### DISCUSSION TITLE
How to download this?

### DISCUSSION THREAD

**Discussion by theridon** [lucidrains/imagen-pytorch#159] [Q&A] [answered]
Forgive as I am new to coding, I just don't get how to install or run this at all. I downloaded and unzipped the contents into a file, there is a "setup.py" file but I have no clue how to open it or proceed with it. Thank you for nay help.

**Comment by ebagjones** [comment]
Frankly, this the equivalent of sitting down at the cockpit of a jumbo jet and asking  where you turn the key and how to honk the horn. 
This is a bleeding edge state of the art AI project that would be difficult for a junior CS graduate to jump into and make meaningful progress with. 

If you are serious about this, go spend a 3-6 months learning Python, 3-6 months doing an ML course or two, and if you are a super fast learner, come back here and see if this project has a trained model ready.

**Reply by theridon** [reply]
No, it's really not. I was referred here as a way to use imagen to generate procedural images. My understanding was, it was similar to Dall-e in that it would be able to take text prompts and turn them into images. If that's not the case, you can just let me know instead of behaving like a pompous ass.

**Reply by ebagjones** [reply]
Well I was trying to be helpful explaining the severe learning curve but it sounds like maybe you meant to ask a very different question based on a misunderstanding of how this technology works.

**Reply by lucidrains** [reply]
yea, you are both right here

on one end it does take some know hows, but i'm trying to automate as much as possible, striking a balance between hobbyist as well as to allow the next generation of phd students around the globe to make a big discovery

@theridon if you are looking for something that is already trained and you just inference on it, you should follow the efforts here https://github.com/LAION-AI/dalle2-laion

**Reply by ebagjones** [reply]
Thanks for all of your hard work on this!

**Comment by Nodja** [comment, marked as answer]
If your goal is to generate images from text then this repo is not able to do so by itself. To run a machine learning model you need two things, the code that trains and runs the model, and the model weights that have been trained on data, usually called a checkpoint file. This repo only contains the code. 

The author of this repo is still in the process of replicating google's paper which this model is based on, I think he's close and in some aspects even surpassed it, but it requires tons of processing power to get a trained model out and validate if the code is correct. To replicate the results in the paper the model needs to be trained of 400M images, a feat like that would roughly take 6-12 months on a single consumer GPU.

In short: It's not ready.

Maybe by the end of the year someone will have trained this model on a huge dataset and released it out to the public.

**Comment by mgatelabs** [comment]
For this repo, you need to wait for someone to build and release a checkpoint, or start training it yourself, which is something I wouldn't want to do.  It might take a while, so as an alternative there are other AI gens that "work", but are basically like going back in time.  For example I've used https://github.com/nerdyrodent/VQGAN-CLIP locally on my rig with the COCO checkpoint, it makes images, but the code is getting dusty.
